File: LeetCode/Easy/0013. Roman To Integer.c
int romanToInt(char *s){
    int n = 0;
    
    while(*s) {
        if(*s == 'I') {
            if(*(s+1) && *(s+1) == 'V') {
                n += 4;
                s+=2;
            } else if (*(s+1) && *(s+1) == 'X') {
                n += 9;
                s+=2;
            } else {
                n += 1;
                s++;
            }
        } else if(*s == 'X') {
            if(*(s+1) && *(s+1) == 'L') {
                n += 40;
                s+=2;
            } else if (*(s+1) && *(s+1) == 'C') {
                n += 90;
                s+=2;
            } else {
                n += 10;
                s++;
            }     
        } else if(*s == 'C') {
            if(*(s+1) && *(s+1) == 'D') {
                n += 400;
                s+=2;
            } else if(*(s+1) && *(s+1) == 'M') {
                n += 900;
                s+=2;
            } else {
                n += 100;
                s++;
            }
        } else if(*s == 'V') { 
            n += 5; s++; 
        } else if(*s == 'L') { 
            n += 50; s++; 
        } else if(*s == 'D') { 
            n += 500; s++; 
        } else if(*s == 'M') { 
            n += 1000; s++; 
        }
    }
    
  
    return n;
}

File: LeetCode/Easy/0013. Roman to Integer.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """

        hmap = {"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000}

        ans = 0

        i = 0
        while i < len(s):
            if i+1 < len(s) and hmap[s[i]] < hmap[s[i+1]]:
                ans = ans - hmap[s[i]] + hmap[s[i+1]]
                i += 2
            else:
                ans += hmap[s[i]]
                i += 1
        
        return ans


        
